evm-contracts/src/v0.6-ovm/AccessControlledAggregator.sol

```diff
@@ -1,217 +1,111 @@
-   * allowed to lapse before allowing an oracle to skip an unfinished round
-   * @param _validator is an optional contract address for validating
-   * external validation of answers
-   * @param _minSubmissionValue is an immutable check for a lower bound of what
-   * submission values are accepted from an oracle
-   * @param _maxSubmissionValue is an immutable check for an upper bound of what
-   * submission values are accepted from an oracle
-   * @param _decimals represents the number of decimals to offset the answer by
-   * @param _description a short description of what is being reported
+   * @notice Creates an AccessControlledAggregator contract.
+   * @dev The underlying aggregator implementation must extend Initializable, as this proxy depends on reserved ([0,50] slots) storage space.
+   * @param implementationAddr The address of underlying aggregator implementation used by the proxy.
    */
-  constructor(
-    address _link,
-    uint128 _paymentAmount,
-    uint32 _timeout,
-    address _validator,
-    int256 _minSubmissionValue,
-    int256 _maxSubmissionValue,
-    uint8 _decimals,
-    string memory _description
-  ) public FluxAggregator(
-    _link,
-    _paymentAmount,
-    _timeout,
-    _validator,
-    _minSubmissionValue,
-    _maxSubmissionValue,
-    _decimals,
-    _description
-  ){}
+  constructor(address implementationAddr) public {
+    implementation = implementationAddr;
+    _setUpGuards();
+  }
 
   /**
-   * @notice get data about a round. Consumers are encouraged to check
-   * that they're receiving fresh data by inspecting the updatedAt and
-   * answeredInRound return values.
-   * @param _roundId the round ID to retrieve the round data for
-   * @return roundId is the round ID for which data was retrieved
-   * @return answer is the answer for the given round
-   * @return startedAt is the timestamp when the round was started. This is 0
-   * if the round hasn't been started yet.
-   * @return updatedAt is the timestamp when the round last was updated (i.e.
-   * answer was last computed)
-   * @return answeredInRound is the round ID of the round in which the answer
-   * was computed. answeredInRound may be smaller than roundId when the round
-   * timed out. answerInRound is equal to roundId when the round didn't time out
-   * and was completed regularly.
-   * @dev overridden funcion to add the checkAccess() modifier
-   * @dev Note that for in-progress rounds (i.e. rounds that haven't yet
-   * received maxSubmissions) answer and updatedAt may change between queries.
+   * @dev Set up guarded functions
    */
-  function getRoundData(uint80 _roundId)
-    public
-    view
-    override
-    checkAccess()
-    returns (
-      uint80 roundId,
-      int256 answer,
-      uint256 startedAt,
-      uint256 updatedAt,
-      uint80 answeredInRound
-    )
-  {
-    return super.getRoundData(_roundId);
+  function _setUpGuards() internal virtual {
```

Comment:
This is now optimized.

Not sure about the EIP-2929, I assume it's not included in the L2 gas metering as the EIP is still in review. But the whole gas metering L2 component is still in development by the Optimism team, that's why they will lunch accepting the `gasPrice=0`.

If EIP-2929 gets included in L1, it will definitely be a factor when solving L2 state disputes on L1. Also, when rolling up L2 state, they will need to pay that extra L1 gas, so it is a factor, but it's not directly connected for example to L2 `ovmSSTORE` operation. In conclusion, there is still a lot of unknowns about OVM gas costs.